File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        stringstream ss;
        string result;
        int i = 0, j = 0;
        while (i < s.length() && j < spaces.size()){
            if (i < spaces[j]){
                ss << s[i];
                i++;
            }
            else{
                ss << " ";
                j++;
            }
        }
        while (i < s.length()){
            ss << s[i];
            i++;
        }
        result = ss.str();
        return result;
    }
};

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        result = []
        i, j = 0, 0
        while i < len(s) and j < len(spaces):
            if i < spaces[j]:
                result.append(s[i])
                i += 1
            else:
                result.append(' ')
                j += 1
        result.append(s[i:])
        return ''.join(result)
        # result = list(s)
        # for i in spaces:
        #     result.insert(i, " ")
        
        # print(result)

        #8876 
        # for space in reversed(spaces):
        #     s = s[:space] +" "+s[space:]
        # return s